R/ff_analyze.R

```diff
@@ -1,130 +1,243 @@
-#' Calculate Scores
-#'
-#' This function calculates scores based on predictions, ground truth, and optional parameters.
+#' Extract date from filename
 #'
-#' @param predictions A character vector or raster object representing the predictions.
-#' @param groundtruth A character vector or raster object representing the ground truth.
-#' @param forestmask An optional character vector or raster object representing the forest mask.
-#' @param csvfile An optional CSV file to which the results will be written.
-#' @param append Logical. If TRUE, results will be appended to the existing CSV file.
-#' @param country Character. If NULL all the overlapping polygons will be processed.
-#' Otherwise the ISO3 code should be given and the analysis_polygons dataset should contain a column called iso3
-#' @param analysis_polygons Optional vector or character vector representing analysis polygons.
-#' @param return_polygons Logical. If TRUE, the polygons with calculated scores will be returned.
-#' @param remove_empty Logical. If TRUE, empty rows (with all scores being zero) will be removed from the output.
-#' @param date character. should be in format (YYYY-MM-DD).
-#' Optional if either groundtruth or predictions is a character to the tiffile.
-#' @param tile character. should be in format AA{N-S}_BBB{W-E}.
-#' Optional if either groundtruth or predictions is a character to the tiffile with a directory name.
-#' @param method character. The shorthand for the method used,
-#' which should also be included in the separate csv file for storing methods
-#' @param addWKT Logical.  Whether the WKT should be added as a column to the CSV output
-#' @param verbose Logical. Whether the steps taken in the function should be verbose.
-#'
-#' @return A vector dataset containing calculated scores for each polygon.
-#'
-#' @export
+#' @param predictions Path to predictions file or SpatRaster object
+#' @param groundtruth Path to groundtruth file or SpatRaster object
+#' @return Character string containing the date in YYYY-MM-DD format
+#' @noRd
+get_date_from_files <- function(predictions, groundtruth) {
```

Comment:
the main function should be on the top of files to keep the structure of R files consistent, so please move the ff_analyze function on the top